File: clients/java/src/main/java/com/themisdb/client/TransactionOptions.java
package com.themisdb.client;

import java.time.Duration;

/**
 * Options for creating a transaction
 */
public class TransactionOptions {
    private IsolationLevel isolationLevel;
    private Duration timeout;

    /**
     * Create default transaction options (READ_COMMITTED isolation)
     */
    public TransactionOptions() {
        this.isolationLevel = IsolationLevel.READ_COMMITTED;
        this.timeout = null;
    }

    /**
     * Create transaction options with specified isolation level
     * 
     * @param isolationLevel Desired isolation level
     */
    public TransactionOptions(IsolationLevel isolationLevel) {
        this.isolationLevel = isolationLevel;
        this.timeout = null;
    }

    /**
     * Create transaction options with isolation level and timeout
     * 
     * @param isolationLevel Desired isolation level
     * @param timeout Transaction timeout
     */
    public TransactionOptions(IsolationLevel isolationLevel, Duration timeout) {
        this.isolationLevel = isolationLevel;
        this.timeout = timeout;
    }

    /**
     * Get the isolation level
     * 
     * @return Isolation level
     */
    public IsolationLevel getIsolationLevel() {
        return isolationLevel;
    }

    /**
     * Set the isolation level
     * 
     * @param isolationLevel Isolation level
     * @return This options object (for chaining)
     */
    public TransactionOptions setIsolationLevel(IsolationLevel isolationLevel) {
        this.isolationLevel = isolationLevel;
        return this;
    }

    /**
     * Get the transaction timeout
     * 
     * @return Timeout duration (null if not set)
     */
    public Duration getTimeout() {
        return timeout;
    }

    /**
     * Set the transaction timeout
     * 
     * @param timeout Timeout duration
     * @return This options object (for chaining)
     */
    public TransactionOptions setTimeout(Duration timeout) {
        this.timeout = timeout;
        return this;
    }
}

File: clients/csharp/ThemisDB.Client/TransactionOptions.cs
/*
╔═════════════════════════════════════════════════════════════════════╗
║ ThemisDB - Hybrid Database System                                   ║
╠═════════════════════════════════════════════════════════════════════╣
  File:            TransactionOptions.cs                              ║
  Version:         0.0.32                                             ║
  Last Modified:   2026-02-23 03:57:07                                ║
  Author:          unknown                                            ║
╠═════════════════════════════════════════════════════════════════════╣
  Quality Metrics:                                                    ║
    • Maturity Level:  🟢 PRODUCTION-READY                             ║
    • Quality Score:   98.0/100                                       ║
    • Total Lines:     39                                             ║
    • Open Issues:     TODOs: 0, Stubs: 1                             ║
╠═════════════════════════════════════════════════════════════════════╣
  Revision History:                                                   ║
    • a629043ab  2026-02-22  Audit: document gaps found - benchmarks and stale annotat... ║
╠═════════════════════════════════════════════════════════════════════╣
  Status: ✅ Production Ready                                          ║
╚═════════════════════════════════════════════════════════════════════╝
 */

namespace ThemisDB.Client;

/// <summary>
/// Options for configuring a transaction
/// </summary>
public class TransactionOptions
{
    /// <summary>
    /// Gets or sets the isolation level for the transaction
    /// </summary>
    public IsolationLevel IsolationLevel { get; set; } = IsolationLevel.ReadCommitted;
    
    /// <summary>
    /// Gets or sets the timeout for the transaction
    /// </summary>
    public TimeSpan? Timeout { get; set; }
}
